app/controllers/shared/api.php

```diff
@@ -292,6 +292,37 @@
                         }
                     }
                 }
+            } elseif($keyType === API_KEY_TEST) {
+                var_dump('keys matches test');
+                // Check if given key match project Test API keys
+                $key = $project->find('secret', $apiKey, 'developmentKeys');
+                if ($key) {
+                    $user = new Document([
+                        '$id' => '',
+                        'status' => true,
+                        'email' => 'app-test.' . $project->getId() . '@service.' . $request->getHostname(),
+                        'password' => '',
+                        'name' => $project->getAttribute('name', 'Untitled'),
+                    ]);
+
+                    $role = Auth::USER_ROLE_ADMIN;
+                    $scopes = $roles[$role]['scopes'];
+
+                    $expire = $key->getAttribute('expire');
+                    if (!empty($expire) && $expire < DateTime::formatTz(DateTime::now())) {
+                        throw new Exception(Exception::PROJECT_KEY_EXPIRED);
+                    }
+
+                    Authorization::setRole(Auth::USER_ROLE_ADMIN);
+                    Authorization::setDefaultStatus(false);  // Cancel security segmentation for API keys.
```

Comment:
Im not sure what development keys are for. Are they like API keys but for Console user - giving access to all project and project/organization actions?

Reply:
They are for debug purpose, ignore rate limits, give detailed error logs etc.